wgpu-hal/src/vulkan/mod.rs

```diff
@@ -1338,6 +1345,17 @@ impl crate::Queue for Queue {
     }
 }
 
+impl Queue {
+    pub fn raw_device(&self) -> &ash::Device {
+        &self.device.raw
+    }
+
+    pub fn add_signal_semaphore(&self, semaphore: vk::Semaphore) {
+        let mut signal_semaphores = self.signal_semaphores.lock();
+        signal_semaphores.push(semaphore);
+    }
```

Comment:
I think it would be reasonable to have an argument that is an Option<u64> for a signal value, and automatically convert None into `!0`

Reply:
Since the value is ignored, initializing with `0` instead of `!0` should be fine too?

Reply:
Ah if it's ignored that's fine. I thought it needed to be !0

Reply:
https://registry.khronos.org/vulkan/specs/latest/man/html/VkTimelineSemaphoreSubmitInfo.html

> If the semaphore in [VkSubmitInfo](https://registry.khronos.org/vulkan/specs/latest/man/html/VkSubmitInfo.html)::pWaitSemaphores or [VkSubmitInfo](https://registry.khronos.org/vulkan/specs/latest/man/html/VkSubmitInfo.html)::pSignalSemaphores corresponding to an entry in pWaitSemaphoreValues or pSignalSemaphoreValues respectively **was not created with a [VkSemaphoreType](https://registry.khronos.org/vulkan/specs/latest/man/html/VkSemaphoreType.html) of VK_SEMAPHORE_TYPE_TIMELINE**, the implementation **must ignore** the value in the pWaitSemaphoreValues or pSignalSemaphoreValues entry.

Funnily enough that's _the entire description_ :)

Reply:
Updated as to take Option for a signal value,